src/runtime_src/core/tools/common/ReportCu.cpp

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright (C) 2020 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "ReportCu.h"
+#include "core/common/query_requests.h"
+#include "core/common/device.h"
+#include "core/common/utils.h"
+
+namespace qr = xrt_core::query;
+
+enum class cu_stat : unsigned short {
+  usage = 0,
+  addr,
+  stat
+};
+
+uint32_t 
+parseComputeUnitStat(const std::vector<std::string>& custat, uint32_t offset, cu_stat kind) 
+{
+  uint32_t ret = 0;
+
+  for (auto& line : custat) {
+    uint32_t ba = 0, cnt = 0, sta = 0;
+#ifdef _WIN32

```

Comment:
We want to refrain from having OS-specific directives. Please use boost::format instead

Reply:
Couldn't find any suitable boost function.